src/amo/components/ShowMoreCard/index.js

```diff
@@ -53,19 +79,38 @@ export class ShowMoreCardBase extends React.Component<InternalProps> {
   contents: HTMLElement | null;
 
   static defaultProps: DefaultProps = {
+    _truncateToMaxHeight: truncateToMaxHeight,
     maxHeight: DEFAULT_MAX_HEIGHT,
   };
 
   componentDidMount() {
-    const { uiState } = this.props;
+    const {
+      _setUIState,
+      _truncateToMaxHeight,
+      maxHeight,
+      setUIState,
+      uiState,
+    } = this.props;
     if (!uiState.readMoreExpanded) {
-      this.truncateToMaxHeight(this.contents);
+      _truncateToMaxHeight({
+        contents: this.contents,
+        maxHeight,
+        setUIState: _setUIState || setUIState,
```

Comment:
We cannot use `defaultProps` to set a default value for `setUIState` because the default is a prop that is injected via the `withUIState` HOC, and is therefore not available when `defaultProps` executes. This is why we need this conditional here and also in `componentDidUpdate`.